File: src/main/java/com/max/reminder/controller/FinancialController.java
package com.max.reminder.controller;

import com.max.reminder.pojo.FinancialRecord;
import com.max.reminder.service.FinancialService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Date;

@RestController
@RequestMapping("/v1/financial/")
public class FinancialController {
    @Autowired
    private FinancialService financialService;

    @PostMapping("insertFinancialRecord")
    public String postBill() {
        FinancialRecord record = new FinancialRecord(new Date(), "test", (int) (0.5f * 100), 5, "test", "This is a test record");
        financialService.insertRecord(record);
        return "success";
    }

    @GetMapping("financialRecords")
    public Object getFinancialRecords() {
        return financialService.getAllFinancialRecords();
    }
}

File: src/main/java/com/max/reminder/controller/FinancialController.kt
package com.max.reminder.controller

import com.max.reminder.common.Enums
import com.max.reminder.pojo.Entities
import com.max.reminder.service.FinancialService
import com.max.reminder.util.DateUtil
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/v1/financial/")
class FinancialController {
    @Autowired
    private lateinit var financialService: FinancialService

    @PostMapping("insertFinancialRecord")
    fun postBill(): String {
        val financial = Entities.Financial(name = "test", DateUtil.date, price = 12.5F, 1, type = Enums.Financial.CASH)
        financialService.insertRecord(financial)
        return "success"
    }

    @GetMapping("financialRecords")
    fun financialRecords(): List<Entities.Financial?>? {
        return financialService.allFinancialRecords()
    }
}